### ISSUE TITLE
Request: Password reset based on username OR e-mail address

### ISSUE THREAD

**Issue by Fossil01** [nZEDb/nZEDb#1732]
I have users who can't rmember which e-mail they signed up with (yes really...). Would be nice to be able to reset your passwording using just you username too.

**Comment by niel** [comment]
Allowing anyone to just specify a username and change the password is a recipe for disaster ;-)

**Comment by Fossil01** [comment]
An e-mail needs to be sent of course. There are a lot of users using catchall or gmail's email+whatever@gmail.com function.

**Comment by niel** [comment]
If an email needs to be sent, why not use the standard "Forgotten Password" functionality?

**Comment by Fossil01** [comment]
Like I said. Some users use catchall. So they will get the e-mail but they just dont remember what they put in front of the @.

**Comment by niel** [comment]
Hmm, I think I understand now. You don't want a script to reset passwords (implied in the question) but a change to the Forgotten Passwords option to use only an user name. Makes sense, I often have the same problem on other sites I use infrequently.

**Comment by Fossil01** [comment]
Yeah. But add it as an extra option, so they should still be able to also use their email address.

**Comment by b3rs3rk** [comment]
@Fossil01 I was looking into this, but thought of potentially a better way.  What if the user could provide their API key instead?  Seems a little more secure and they probably can go to two or three different places for it at any given time.  And it prevents another user from spamming someone else with reset requests.

**Comment by ghost** [comment]
API key would be fine too. Need just another method for stupid users forgetting their fake e-mail which they signup with :P
